### PR TITLE
docs(path): improve API docs

### PR DESCRIPTION
**Pull request by kt3k**
part of #4600 #3764

supercedes #4839

### CODE REVIEW

File: path/basename.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import { basename as windowsBasename } from "./windows/basename.ts";
  *
  * @param path - path to extract the name from.
  * @param [suffix] - suffix to remove from extracted name.
```
Comment (iuioiua) [inline path/basename.ts:22]: These nits appear throughout the package.
```suggestion
 * @param path Path to extract the name from.
 * @param suffice Suffix to remove from extracted name.
```

File: path/basename.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ import { basename as windowsBasename } from "./windows/basename.ts";
...
+ * @returns basename of the path.
```
Comment (iuioiua) [inline path/basename.ts:23]: Nit: proper casing
```suggestion
 * @returns The basename of the path.
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
I see quite a few instances of improper casing and partial sentences. Can you please fix those?

File: path/basename.ts [hunk 3]
```diff
@@ -6,20 +6,27 @@ import { basename as posixBasename } from "./posix/basename.ts";
...
  *
- * basename("/home/user/Documents/"); // "Documents"
- * basename("C:\\user\\Documents\\image.png"); // "image.png"
- * basename("/home/user/Documents/image.png", ".png"); // "image"
+ * if (Deno.build.os === "windows") {
+ *   assertEquals(basename("C:\\user\\Documents\\image.png"), "image.png");
+ * } else {
+ *   assertEquals(basename("/home/user/Documents/image.png"), "image.png");
+ * }
  * ```
```
Comment (iuioiua) [inline path/basename.ts:24]: What if we wrote the OS-dependent functions like this? It clearly shows the differences in behaviors between difference OS systems.

**Review by iuioiua — CHANGES_REQUESTED** [review]
I've fixed quite a few nits and added a few improvements. Though, I think there's a bit more work to do here:
1. Mention the differences in behavior for OS-dependent APIs.
2. Add `../path/posix/mod.ts` and `../path/windows/mod.ts` to the Super Linter, and document everything in those directories. We should get top-level documentation to a good state, then just copy the relevant details over.
3. Improve the example code snippets (see suggestion).

File: path/basename.ts (hunk 3 above)
Comment (kt3k) [reply path/basename.ts:24]: I'm not sure it's a good idea to use Deno specific API in examples as we specify this package as cross-runtime.

Comment (iuioiua) [reply path/basename.ts:24]: Does `isWindows` evaluate to the correct OS on Node?

Comment (kt3k) [reply path/basename.ts:24]: Currently no. But I think we should add that support.

Comment (iuioiua) [reply path/basename.ts:24]: I agree; we should add that support. But in the meantime, this package is only partially cross-platform and we should remove the other runtimes from the displayed list on JSR until it's truly cross-platform. I think it'd be good to have the assertions in these top-level functions, for now, until that's solved, as they guarantee correctness. My opinion isn't strong enough to block this PR. Up to you.

Comment (iuioiua) [reply path/basename.ts:24]: See #4914

Comment (kt3k) [reply path/basename.ts:24]: Removed Node and Bun compatibility on JSR.

Added assertions to examples.

**Comment by 91pornaaaaa** [comment]
https://91porna.com/

**Review by iuioiua — APPROVED** [review]
Big win! Great work!

**Pull request merged**